app/layout.tsx

```diff
@@ -1,50 +1,53 @@
-import type { Metadata } from "next";
-import { Inter } from "next/font/google";
-import "./globals.css";
-import dynamic from "next/dynamic";
-import { Suspense } from "react";
-import getServerClient from "@/utils/ld-server/serverClient";
-import serverflag from "@/utils/ld-server/flaggetter";
-import { unstable_noStore as noStore } from "next/cache";
-import { LDClient } from "launchdarkly-node-server-sdk";
-
-const inter = Inter({ subsets: ["latin"] });
-
-const AsyncLDProvider = dynamic(() => import("@/components/ldprovider"), {
-  ssr: false,
-});
+import { getNodeClient } from '@/launchDarkly/server';
+import type { Metadata } from 'next';
+import { unstable_noStore as noStore } from 'next/cache';
+import dynamic from 'next/dynamic';
+import { Inter } from 'next/font/google';
+import { Suspense } from 'react';
+
+import './globals.css';
+
+const inter = Inter({ subsets: ['latin'] });
 
 export const metadata: Metadata = {
-  title: "NextJS 14 + LaunchDarkly Template (Unofficial)",
-  description: "NextJS 14 and LaunchDarkly template for quick scaffolding",
+  title: 'NextJS 14 + LaunchDarkly Template (Unofficial)',
+  description: 'NextJS 14 and LaunchDarkly template for quick scaffolding',
 };
 
+const LDProvider = dynamic(() => import('@/launchDarkly/client').then((c) => c.LDProvider), {
+  ssr: false,
+});
+
 export default async function RootLayout({
   children,
 }: Readonly<{
   children: React.ReactNode;
 }>) {
-
   // Prevents caching of the page which results in static feature flag values at build time
-  noStore()
+  noStore();
 
+  // The context is usually sourced from the request cookie. This is hardcoded for example only.
   const context = {
-    kind: "user",
-    key: "1",
-    name: "Anonymous",
+    kind: 'user',
+    key: '1',
   };
 
-  // Server SDK Client
-  // const client: LDClient = await getServerClient(process.env.LAUNCHDARKLY_SDK_KEY!);
+  const nodeClient = getNodeClient();
+  const bootstrap = await nodeClient.allFlagsState(context);
 
   // Sample flag getter
-  // const flag = await serverflag(client, "flag-key", context, false);
+  // const flag = await nodeClient.variation('flag-key', context, false);
 
   return (
     <html lang="en">
       <body className={inter.className}>
         <Suspense fallback={<div>Loading...</div>}>
-          <AsyncLDProvider context={context}>{children}</AsyncLDProvider>
+          <LDProvider
+            context={context}
+            options={{ bootstrap: JSON.parse(JSON.stringify(bootstrap)) }}
```

Comment:
This works! You can test by inspecting the network tab on the browser debugger and see there's no initial call for flags.

Reply:
This is so awesome and going to be such a low effort performance boost. Great thing is that the all flags call doesn't trigger evaluation events either, so theres no hit on experimentation.

We should double check with them though for client side experiments that get the values first from this method, is it still going to track events successfully? I think yes, just want to confirm. @mike-zorn or @cdelst?

Reply:
On the client side, `useFlags` invokes `.variation` which tracks events correctly. I think we're good here.